invenio_rdm_records/oaiserver/services/services.py

```diff
@@ -38,7 +37,7 @@ class OAIPMHServerService(Service):
     def __init__(self, config, extra_reserved_prefixes={}):
         """Init service with config."""
         super().__init__(config)
-        self.reserved_prefixes = config.reserved_prefixes.union(extra_reserved_prefixes)
+        self.extra_reserved_prefixes = extra_reserved_prefixes
```

Comment:
Having an object as a default value will use the same object for all instances of this class. Better to have `extra_reserved_prefixes` defined as `None` and then initializing like `self.extra_reserved_prefixes = extra_reserved_prefixes or {}`